pkg/skaffold/runner/v1/new.go

```diff
@@ -152,6 +153,7 @@ func NewForConfig(runCtx *runcontext.RunContext) (*SkaffoldRunner, error) {
 		runCtx:             runCtx,
 		intents:            intents,
 		isLocalImage:       isLocalImage,
+		localDeploy:        local,
```

Comment:
we should think about `multi-config` projects having both `k8s` and `docker` deployers. Otherwise it might be difficult to remove assumptions and checks like these later on if we don't keep it in scope from the get-go.

Reply:
https://github.com/GoogleContainerTools/skaffold/pull/6124 should actually make it easy to support this case in skaffold, but I'm still not convinced this is a valid use case. I can't really think of a scenario where you'd want to deploy part of your application to a k8s cluster, and then another part of it as a single container running in your local daemon as these would very likely not be able to talk to each other.

for now, I think we can get away with doing nothing (since now there will be nothing in skaffold preventing you from doing this), but moving forward we'll want to explore this use case and determine if there are real scenarios we want to support.